File: backend/src/main/java/com/tripfriend/global/annotation/CheckPermission.java
package com.tripfriend.global.annotation;

import java.lang.annotation.*;

@Retention(RetentionPolicy.RUNTIME) // 런타임까지 유지
@Target(ElementType.METHOD) // 메서드에만 적용 가능
public @interface CheckPermission {
    String value(); // 권한 (ex: "ADMIN", "USER")
}

File: backend/src/main/java/com/tripfriend/global/annotation/CheckPermission.kt
package com.tripfriend.global.annotation

@Retention(AnnotationRetention.RUNTIME) // 런타임까지 유지
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER) // 메서드에만 적용 가능
annotation class CheckPermission(
    val value: String // 권한 (ex: "ADMIN", "USER")
)
